### PR TITLE
[video-player] add support for content uris as urls

### PR DESCRIPTION
**Pull request by udnisap**
#1706  Description

content://  uris will be routed from DefaultDataSourceFactory in Android

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by udnisap** [comment]
Does it make sense to flip the condition here (https://github.com/flutter/plugins/blob/f54590ffac5b98fe2f046cec1c47306d15358f27/packages/video_player/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java#L84) to use DefaultDataSourceFactory by default and  DefaultHttpDataSourceFactory on http or https?

**Comment by cyanglaz** [comment]
> https

I think you are right. We can get rid of the `isFileOrAsset` and make it something like `isHTTP`. Do you mind to do this change?

File: packages/video_player/example/lib/main.dart [hunk 1]
```diff
@@ -386,11 +386,11 @@ void main() {
                     Container(
                       padding: const EdgeInsets.only(top: 20.0),
                     ),
-                    const Text('With remote m3u8'),
+                    const Text('With remote mp4'),
```
Comment (cyanglaz) [inline packages/video_player/example/lib/main.dart]: why did we change this and the url below?

File: packages/video_player/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+## 0.10.1+6
+
+* Use DefaultHttpDataSourceFactory only when network schemas and user
```
Comment (cyanglaz) [inline packages/video_player/CHANGELOG.md]: nits: `Use DefaultHttpDataSourceFactory only for network schemas and use`

File: packages/video_player/example/lib/main.dart (hunk 1 above)
Comment (udnisap) [reply packages/video_player/example/lib/main.dart]: that URL was not accessible in public network. was it an internal URL? 
i tried with the URL on the docs and updated it. i can revert it back

Comment (udnisap) [reply packages/video_player/example/lib/main.dart]: reverted back to m3u8

**Comment by udnisap** [comment]
@cyanglaz does everything looks good to you?

**Pull request merged**